pylzy/lzy/env/explorer/utils.py

```diff
@@ -97,20 +97,82 @@ def get_builtin_module_names() -> FrozenSet[str]:
     return frozenset(sys.builtin_module_names)
 
 
-@lru_cache(maxsize=None)  # cache size is about few MB
-def get_files_to_distributions() -> Dict[str, Distribution]:
-    result = {}
-
+@lru_cache(maxsize=None)
+def get_names_to_distributions() -> Dict[str, Distribution]:
+    result: Dict[str, Distribution] = {}
     # NB: importlib_metadata.Distribution calls may return different
     # results in depends from cwd.
     # So we are moving cwd into tmp dir in name of results repeatability.
     # In case of PermissionError, location of tmp dir can be moved with
     # TMPDIR env variable.
-    with TemporaryDirectory() as tmp:
-        with change_working_directory(tmp):
-            for distribution in importlib_metadata.distributions():
-                for filename in distribution.files or ():
-                    fullpath = distribution.locate_file(filename)
-                    result[str(fullpath)] = distribution
+    with tmp_cwd():
+        for distribution in importlib_metadata.distributions():
+            result[distribution.name] = distribution
+
+    return result
+
+
+@lru_cache(maxsize=None)  # cache size is about few MB
+def get_files_to_distributions() -> Dict[str, Distribution]:
+    result = {}
+
+    for distribution in get_names_to_distributions().values():
+        # NB: TODO: distribution.files could be empty in case of
+        # dist-packages & .egg-info.
+        # Primarily it affects python packages, installed
+        # via apt-get, for example.
+        # So if user working in non-isolated environment
+        # and using packages from dist-packages, we will
+        # mistreat these packages as local in future while
+        # classification.
+        for filename in distribution.files or ():
+            fullpath = distribution.locate_file(filename)
+            result[str(fullpath)] = distribution
+
+    return result
+
+
+def check_distribution_is_meta_package(distribution: Distribution) -> bool:
+    if not distribution.files:
+        # NB: .egg-info packages and some apt-packages doensn't have
+        # a dist-info directory and importlib.metadata fails to generate
+        # files list
+        return False
+
+    for filename in distribution.files:
```

Comment:
Actually we need to check that distribution contains only `.dist-info` or `.egg-info` (or both) directories. Current code with loop with double negations inside is less understandable than it could be.

I suppose this is more readable:
```py
directories = [Path(filename).parts[0] for filename in distribution.files]
return all(directory.endswith('.dist-info') or directory(endswith('.egg-info') for directory in directories)
```